src/tests/functional/shared_test_classes/src/single_op/multiclass_nms.cpp

```diff
@@ -113,6 +114,202 @@ void MulticlassNmsLayerTest::generate_inputs(const std::vector<ov::Shape>& targe
     }
 }
 
+
+void MulticlassNmsLayerTest::GetOutputParams(size_t& numBatches, size_t& maxOutputBoxesPerBatch) {
+    const auto& funcInputs = function->inputs();
+    const auto& boxes_dims = inputs[funcInputs[0].get_node_shared_ptr()].get_shape();
+    const auto& scores_dims = inputs[funcInputs[1].get_node_shared_ptr()].get_shape();
+
+    const auto shared = (scores_dims.size() == 3);
+    if (!shared)
+        ASSERT_TRUE(funcInputs.size() > 2) << "Expected 3 inputs when input 'score' is 2D.";
+
+    const auto& roisnum_dims = funcInputs.size() >2 ? inputs[funcInputs[2].get_node_shared_ptr()].get_shape() : Shape();
+
+    const auto numBoxes = shared ? boxes_dims[1] : boxes_dims[1];
+    auto numClasses = shared ? scores_dims[1] : boxes_dims[0];
+    numBatches = shared ? scores_dims[0] : roisnum_dims[0];
+
+    ASSERT_TRUE(numBatches > 0 && numBoxes > 0 && numClasses > 0)
+        << "Expected numBatches, numBoxes, numClasses > 0, got:" << numBatches << ", " << numBoxes << ", " << numClasses;
+
+    auto realClasses = numClasses;
+    if (m_attrs.background_class >= 0 && m_attrs.background_class < numClasses) {
+       realClasses = realClasses - 1;
+    }
+
+    size_t maxOutputBoxesPerClass = 0;
+    if (m_attrs.nms_top_k >= 0)
+       maxOutputBoxesPerClass = std::min(numBoxes, static_cast<size_t>(m_attrs.nms_top_k));
+    else
+       maxOutputBoxesPerClass = numBoxes;
+
+    maxOutputBoxesPerBatch  = maxOutputBoxesPerClass * realClasses;
+    if (m_attrs.keep_top_k >= 0)
+       maxOutputBoxesPerBatch =
+               std::min(maxOutputBoxesPerBatch, static_cast<size_t>(m_attrs.keep_top_k));
+}
+
+void MulticlassNmsLayerTest::compare(const std::vector<ov::Tensor> &expectedOutputs,
+                                     const std::vector<ov::Tensor> &actualOutputs) {
+    if (targetDevice != ov::test::utils::DEVICE_GPU) {
+        SubgraphBaseTest::compare(expectedOutputs, actualOutputs);
+        return;
+    }
```

Comment:
I would prefer to move such huge custom logic to a separate GPU-specific class inherited from base class. It becomes too complicated for maintenance

Reply:
And moreover, we have to avoid a device specific logic in shared lib